src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IInterpolatedStringExpression.vb

```diff
@@ -423,5 +423,820 @@ BC30109: '[Class]' is a class type and cannot be used as an expression.
+    Public Sub M(a As Boolean, b As Integer, c As Integer, d As String, p As String)'BIND:"Public Sub M(a As Boolean, b As Integer, c As Integer, d As String, p As String)"
+        p = $"{d,If(a, b, c):D3}"
+    End Sub
+End Class]]>.Value
+
+            Dim expectedFlowGraph = <![CDATA[
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: 'p = $"{d,If ...  b, c):D3}"')
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.String, IsInvalid, IsImplicit) (Syntax: 'p = $"{d,If ...  b, c):D3}"')
+              Left: 
+                IParameterReferenceOperation: p (OperationKind.ParameterReference, Type: System.String) (Syntax: 'p')
+              Right: 
+                IInterpolatedStringOperation (OperationKind.InterpolatedString, Type: System.String, IsInvalid) (Syntax: '$"{d,If(a, b, c):D3}"')
+                  Parts(2):
+                      IInterpolationOperation (OperationKind.Interpolation, Type: null, IsInvalid) (Syntax: '{d,')
+                        Expression: 
+                          IParameterReferenceOperation: d (OperationKind.ParameterReference, Type: System.String) (Syntax: 'd')
+                        Alignment: 
+                          ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0, IsInvalid) (Syntax: '')
+                        FormatString: 
+                          null
+                      IInterpolatedStringTextOperation (OperationKind.InterpolatedStringText, Type: null, IsInvalid) (Syntax: 'If(a, b, c):D3')
+                        Text: 
+                          ILiteralOperation (OperationKind.Literal, Type: System.String, Constant: "If(a, b, c):D3", IsInvalid, IsImplicit) (Syntax: 'If(a, b, c):D3')
+
+    Next (Regular) Block[B2]
+Block[B2] - Exit
+    Predecessors: [B1]
+    Statements (0)
+]]>.Value
+
+            Dim expectedDiagnostics = <![CDATA[
+BC30204: Integer constant expected.
+        p = $"{d,If(a, b, c):D3}"
+                 ~
+BC30370: '}' expected.
+        p = $"{d,If(a, b, c):D3}"
+                 ~
+BC30035: Syntax error.
+        p = $"{d,If(a, b, c):D3}"
+                               ~
+]]>.Value
+
+            VerifyFlowGraphAndDiagnosticsForTest(Of MethodBlockSyntax)(source, expectedFlowGraph, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)>
+        <Fact>
+        Public Sub InterpolatedStringExpression_ConditionalCodeInAlignment_Flow_02()
+            Dim source = <![CDATA[
+Imports System
+
+Friend Class [Class]
+    Public Sub M(a As Boolean, b As String, c As String, c2 As String, p As String)'BIND:"Public Sub M(a As Boolean, b As String, c As String, c2 As String, p As String)"
+        p = $"{c2,If(a, b, c):D3}"
```

Comment:
>If(a, b, c) [](start = 18, length = 11)

I am curious would it become conditional expression if you parenthesize it? If it would, consider adjusting this test like that.

Reply:
Sure, will try it out and update the test.

Reply:
Verified it by adjusting the test, and it doesn't. I will retain the original test.

---
In reply to: [186187613](https://github.com/dotnet/roslyn/pull/26545#discussion_r186187613) [](ancestors = 186187613)